module.tf

```diff
@@ -0,0 +1,136 @@
+    dynamic "backup_configuration" {
+      for_each = var.backup_configuration
+      content {
+        binary_log_enabled = lookup(backup_configuration.value, "binary_log_enabled", null)
+        enabled            = lookup(backup_configuration.value, "enabled", null) #data.null_data_source.gcp_postgres_tags[count.index].inputs["application"] == "central" ? false : lookup(backup_configuration.value, "enabled", null)
+        start_time         = lookup(backup_configuration.value, "start_time", null)
+        location           = lookup(backup_configuration.value, "location", null) #var.central_backup_location == null &&  data.null_data_source.gcp_postgres_tags[count.index].inputs["application"] == "central" ? var.central_backup_location : 
+        point_in_time_recovery_enabled = lookup(backup_configuration.value, "point_in_time_recovery_enabled", null)
+        backup_retention_settings  {
+          retained_backups  = lookup(backup_configuration.value, "retained_backups", 35)
+          retention_unit   = lookup(backup_configuration.value, "retention_unit","COUNT")
+        }
+      }
+    }
+    dynamic "insights_config" { 
+    for_each = var.insights_config != [] ? var.insights_config : []
+    content {
+            query_insights_enabled = lookup(insights_config.value, "query_insights_enabled", true)
+            query_string_length = lookup(insights_config.value, "query_string_length", 1024)  
+            record_application_tags= lookup(insights_config.value, "record_application_tags", false) 
+            record_client_address= lookup(insights_config.value, "record_client_address", false)
+    }
+    }
+    dynamic "maintenance_window" {
+      for_each = var.maintenance_window
+      content {
+        day          = lookup(maintenance_window.value, "day", null)
+        hour         = lookup(maintenance_window.value, "hour", null)
+        update_track = lookup(maintenance_window.value, "update_track", null)
+      }
+    }
+
+    ip_configuration {
+      require_ssl      = lookup(var.postgres_dbs[count.index],"require_ssl",false)
+      ipv4_enabled = var.postgres_dbs[count.index]["ipv4_enabled"]
+      private_network = var.private_network
+      dynamic "authorized_networks" {
+        for_each = var.postgres_dbs[count.index]["ipv4_enabled"] == true ? concat(split(";", var.natgw_ip), var.authorized_networks_list) : []
+        iterator = authorized_networks_list
+
+        content {
+          name  = "allow-${authorized_networks_list.key}"
+          value = authorized_networks_list.value
+        }
+      }
+    }
+    dynamic "database_flags" {
+      for_each = var.postgres_dbs[count.index]["database_flags_map"]
+      iterator = database_flags_map
+
+      content {
+        name  = database_flags_map.key
+        value = database_flags_map.value
+      }
+    }
+  }
+  lifecycle {
+    ignore_changes = [
+      #settings.0.replication_type, 
+      settings.0.disk_size ##should be remove in 3 weeks##
+      ]
+  }
+}
```

Comment:
[comment]: <> (FrogbotReviewComment)

## 🛠️ Infrastructure as Code

<div align="center">

| Severity | Finding |
| :--------------: | :---: |
| ![](https://raw.githubusercontent.com/jfrog/frogbot/master/resources/v2/applicableHighSeverity.png)<br>    High | No TLS for CloudSQL was detected |

</div>

<details>
<summary> <b>Full description</b> </summary>
<br>

Resource `google_sql_database_instance` should have `settings.ip_configuration.require_ssl=true`

Vulnerable example - 
```
resource "google_sql_database_instance" "vulnerable_example" {
    settings {
        ip_configuration {
            require_ssl = false  # or unset
        }
    }
 }
```

Secure example -
```
resource "google_sql_database_instance" "secure_example" {
    settings {
        ip_configuration {
            require_ssl = true
        }
    }
 }
```

</details>


---
<div align="center">

[🐸 JFrog Frogbot](https://github.com/jfrog/frogbot#readme)

</div>